### PR TITLE
Construct embedded weaviate download URL from version

### PR DESCRIPTION
**Pull request by dirkkul**
allows:
```
client = weaviate.Client(embedded_options=EmbeddedOptions(version="latest"))
```

and
```
client = weaviate.Client(embedded_options=EmbeddedOptions(version="v1.18.2"))
```

### CODE REVIEW

File: weaviate/embedded.py [hunk 1]
```diff
@@ -46,11 +49,42 @@ def __init__(self, options: EmbeddedOptions):
         self.ensure_paths_exist()
         self.check_supported_platform()
         self._parsed_weaviate_version = ""
+        # regular expression to detect a version number: v[one digit].[1-2 digits].[1-2 digits] - nothing in front or back
+        pattern = re.compile(r"^v\d\.\d{1,2}\.\d{1,2}$")
```
Comment (parkerduckworth) [inline weaviate/embedded.py:53]: I think the `v` prefix is redundant and can be omitted.

Also, restricting to a suffix of `[1-2 digits] - nothing in front or back` prevents the use of beta or alpha releases.

Comment (dirkkul) [reply weaviate/embedded.py]: removed the "v", I also added the possibility of adding a "-rc.X" tag. We never had alpha or beta releases for Weaviate, so I left those out

File: weaviate/embedded.py [hunk 2]
```diff
@@ -49,15 +51,23 @@ def __init__(self, options: EmbeddedOptions):
...
-        pattern = re.compile(r"^v\d\.\d{1,2}\.\d{1,2}$")
-        if self.options.version.startswith(GITHUB_RELEASE_DOWNLOAD_URL):
-            # replace with str.removeprefix() after 3.8 has been deprecated
-            self._parsed_weaviate_version = self.options.version[
-                len(GITHUB_RELEASE_DOWNLOAD_URL) :
-            ].split("/")[0]
+        # regular expression to detect a version number: v[one digit].[1-2 digits].[1-2 digits]
+        # optionally there can be a "-rc.[1-2 digits]"
+        # nothing in front or back
+        version_pattern = re.compile(r"^\d\.\d{1,2}\.\d{1,2}?(-rc\.\d{1,2}|$)$")
```
Comment (parkerduckworth) [inline weaviate/embedded.py:57]: I think the regex pattern should be a bit broader still, as we historically have had `alpha` releases in addition to `rc` releases:

https://github.com/weaviate/weaviate/tree/v1.18.0-alpha.1

File: weaviate/embedded.py (hunk 1 above)
Comment (parkerduckworth) [reply weaviate/embedded.py]: See [this comment](https://github.com/weaviate/weaviate-python-client/pull/281#discussion_r1158733405) about alpha releases

File: weaviate/embedded.py (hunk 2 above)
Comment (dirkkul) [reply weaviate/embedded.py]: ah, I only checked the release page. Added alpha and beta

**Review by parkerduckworth — APPROVED** [review]

**Pull request merged**